src/plugins/columnSorting/columnSorting.js

```diff
@@ -154,30 +152,52 @@ class ColumnSorting extends BasePlugin {
    * @fires Hooks#beforeColumnSort
    * @fires Hooks#afterColumnSort
    */
-  sort(column, order) {
-    this.setSortingColumn(column, order);
+  sort(column, order = this.getNextOrderState(column)) {
+    const allowSorting = this.hot.runHooks('beforeColumnSort', column, order);
 
-    if (isUndefined(this.sortColumn)) {
+    if (isUndefined(column)) {
       return;
     }
 
-    const allowSorting = this.hot.runHooks('beforeColumnSort', this.sortColumn, this.sortOrder);
-
     if (allowSorting === false) {
       return;
     }
 
+    this.sortColumn = this.hot.toPhysicalColumn(column);
+    this.sortOrder = order;
+
     this.sortByPresetColumnAndOrder();
     this.updateSortIndicator();
 
-    this.hot.runHooks('afterColumnSort', this.sortColumn, this.sortOrder);
+    this.hot.runHooks('afterColumnSort', column, order);
 
     this.hot.render();
     this.hot.view.wt.draw(true);
 
     this.saveSortingState();
   }
 
+  /**
+   * Get new order state for particular column. The states queue looks as follows: 'asc' -> 'desc' -> 'none' -> 'asc'
+   *
+   * @param {Number} column Visual column index.
+   * @returns {String} Sorting order (`asc` for ascending, `desc` for descending and `none` for initial state).
+   */
+  getNextOrderState(column) {
+    const physicalColumn = this.hot.toPhysicalColumn(column);
+
+    if (this.sortColumn === physicalColumn) {
+      if (this.sortOrder === DESC_SORT_STATE) {
+        return NONE_SORT_STATE;
+
+      } else if (this.sortOrder === ASC_SORT_STATE) {
+        return DESC_SORT_STATE;
+      }
+    }
+
+    return ASC_SORT_STATE;
```

Comment:
Could you use `return` once? With the cached result in a variable.
@budnix mentioned that will be useful when we implement automatic code coverage.

Reply:
For discussion with @jansiegel. I will remember that.